treeherder/model/management/commands/cycle_data.py

```diff
@@ -320,6 +321,83 @@ def __lookup_new_signature(self):
             raise LookupError('Exhausted all signatures originating from try repository.')
 
 
+class IrrelevantDataRemoval(RemovalStrategy):
+    """
+    Removes `performance_datum` rows
+    that originate from repositories other than
+    `autoland`, `mozilla-central`, `fenix`, `reference-browser` and
+    `mozilla-beta` that are more than 6 months old.
+    """
+
+    repositories = None
+    repositories_ids = []
+    repository_names = [
+        'autoland',
+        'mozilla-central',
+        'mozilla-beta',
+        'fenix',
+        'reference-browser',
+    ]
+
+    def __init__(self, chunk_size: int):
+        self._cycle_interval = timedelta(days=(6 * 30))
+        self._chunk_size = chunk_size
+        self._max_timestamp = datetime.now() - self._cycle_interval
+        self._manager = PerformanceDatum.objects
+
+    @property
+    def relevant_repositories(self):
```

Comment:
This could return a list of ids. We don't really need a list of repository ORM objects.